diff --git tensorflow_federated/cc/core/impl/aggregation/core/BUILD tensorflow_federated/cc/core/impl/aggregation/core/BUILD
index 80bb0b9ae..9c60975fc 100644
--- tensorflow_federated/cc/core/impl/aggregation/core/BUILD
+++ tensorflow_federated/cc/core/impl/aggregation/core/BUILD
@@ -12,8 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-load("@com_github_grpc_grpc//bazel:python_rules.bzl", "py_proto_library")
 load("@rules_cc//cc:defs.bzl", "cc_proto_library")
+load("@rules_python//python:proto.bzl", "py_proto_library")
 
 package(
     default_applicable_licenses = ["//:package_license"],
diff --git tensorflow_federated/cc/core/impl/aggregation/protocol/BUILD tensorflow_federated/cc/core/impl/aggregation/protocol/BUILD
index cbe8cada5..aa9fb25d7 100644
--- tensorflow_federated/cc/core/impl/aggregation/protocol/BUILD
+++ tensorflow_federated/cc/core/impl/aggregation/protocol/BUILD
@@ -12,8 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-load("@com_github_grpc_grpc//bazel:python_rules.bzl", "py_proto_library")
 load("@rules_cc//cc:defs.bzl", "cc_proto_library")
+load("@rules_python//python:proto.bzl", "py_proto_library")
 
 package(
     default_applicable_licenses = ["//:package_license"],
diff --git tensorflow_federated/cc/core/impl/aggregation/tensorflow/BUILD tensorflow_federated/cc/core/impl/aggregation/tensorflow/BUILD
index 9fdc5fadd..7d23a5940 100644
--- tensorflow_federated/cc/core/impl/aggregation/tensorflow/BUILD
+++ tensorflow_federated/cc/core/impl/aggregation/tensorflow/BUILD
@@ -33,7 +33,7 @@ cc_library(
         "@org_tensorflow//tensorflow/core:framework",
         "@org_tensorflow//tensorflow/core:protos_all_cc",
         "@org_tensorflow//tensorflow/core/platform:tstring",
-        "@org_tensorflow//tensorflow/tsl/platform:refcount",
+        "@tsl//tsl/platform:refcount",
     ],
 )
 
@@ -133,7 +133,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:cord",
         "@org_tensorflow//tensorflow/core/platform:env",
-        "@org_tensorflow//tensorflow/tsl/platform:env",
+        "@tsl//tsl/platform:env",
     ],
 )
 
@@ -168,7 +168,7 @@ cc_library(
         "@com_google_absl//absl/strings",
         "@com_google_absl//absl/strings:cord",
         "@org_tensorflow//tensorflow/core/platform:env",
-        "@org_tensorflow//tensorflow/tsl/platform:env",
+        "@tsl//tsl/platform:env",
     ],
 )
 
diff --git tensorflow_federated/cc/core/impl/aggregation/tensorflow/converters.cc tensorflow_federated/cc/core/impl/aggregation/tensorflow/converters.cc
index 3db0cab4f..098bae0c5 100644
--- tensorflow_federated/cc/core/impl/aggregation/tensorflow/converters.cc
+++ tensorflow_federated/cc/core/impl/aggregation/tensorflow/converters.cc
@@ -29,7 +29,7 @@
 #include "tensorflow/core/framework/tensor_shape.h"
 #include "tensorflow/core/framework/tensor_shape.pb.h"
 #include "tensorflow/core/platform/tstring.h"
-#include "tensorflow/tsl/platform/refcount.h"
+#include "tsl/platform/refcount.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/base/monitoring.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/core/datatype.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/core/tensor.h"
diff --git tensorflow_federated/cc/core/impl/aggregation/tensorflow/tensorflow_checkpoint_builder_factory.cc tensorflow_federated/cc/core/impl/aggregation/tensorflow/tensorflow_checkpoint_builder_factory.cc
index 3903a74d3..1febef402 100644
--- tensorflow_federated/cc/core/impl/aggregation/tensorflow/tensorflow_checkpoint_builder_factory.cc
+++ tensorflow_federated/cc/core/impl/aggregation/tensorflow/tensorflow_checkpoint_builder_factory.cc
@@ -29,7 +29,7 @@
 #include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 #include "tensorflow/core/platform/file_system.h"
-#include "tensorflow/tsl/platform/env.h"
+#include "tsl/platform/env.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/base/monitoring.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/core/tensor.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/protocol/checkpoint_builder.h"
diff --git tensorflow_federated/cc/core/impl/aggregation/tensorflow/tensorflow_checkpoint_parser_factory.cc tensorflow_federated/cc/core/impl/aggregation/tensorflow/tensorflow_checkpoint_parser_factory.cc
index dd2195773..ce736d63b 100644
--- tensorflow_federated/cc/core/impl/aggregation/tensorflow/tensorflow_checkpoint_parser_factory.cc
+++ tensorflow_federated/cc/core/impl/aggregation/tensorflow/tensorflow_checkpoint_parser_factory.cc
@@ -29,7 +29,7 @@
 #include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 #include "tensorflow/core/platform/file_system.h"
-#include "tensorflow/tsl/platform/env.h"
+#include "tsl/platform/env.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/base/monitoring.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/core/tensor.h"
 #include "tensorflow_federated/cc/core/impl/aggregation/protocol/checkpoint_parser.h"
diff --git tensorflow_federated/cc/core/impl/executor_stacks/executor_stack_bindings.cc tensorflow_federated/cc/core/impl/executor_stacks/executor_stack_bindings.cc
index 6f6253b4a..2f97a2226 100644
--- tensorflow_federated/cc/core/impl/executor_stacks/executor_stack_bindings.cc
+++ tensorflow_federated/cc/core/impl/executor_stacks/executor_stack_bindings.cc
@@ -17,7 +17,7 @@ limitations under the License
 #include <memory>
 #include <vector>
 
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "include/pybind11/detail/common.h"
 #include "include/pybind11/pybind11.h"
 #include "include/pybind11/stl.h"
diff --git tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.cc tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.cc
index 430f16856..229dc5362 100644
--- tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.cc
+++ tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.cc
@@ -27,7 +27,7 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/str_cat.h"
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/composing_executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.h tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.h
index 6b5e033db..eca439e0a 100644
--- tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.h
+++ tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.h
@@ -22,7 +22,7 @@ limitations under the License
 #include <vector>
 
 #include "absl/status/statusor.h"
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/composing_executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks_test.cc tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks_test.cc
index 6ec2e8373..3b188a3e4 100644
--- tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks_test.cc
+++ tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks_test.cc
@@ -24,9 +24,9 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "include/grpc/support/time.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/impl/call.h"
-#include "include/grpcpp/security/credentials.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/impl/call.h"
+#include "grpcpp/security/credentials.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/composing_executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/BUILD tensorflow_federated/cc/core/impl/executors/BUILD
index 3a90cf353..b1f17e8ee 100644
--- tensorflow_federated/cc/core/impl/executors/BUILD
+++ tensorflow_federated/cc/core/impl/executors/BUILD
@@ -328,7 +328,7 @@ cc_library(
         "@com_google_absl//absl/types:span",
         "@com_google_absl//absl/utility",
         "@federated_language//federated_language/proto:computation_cc_proto",
-        "@org_tensorflow//tensorflow/tsl/profiler/lib:traceme",
+        "@tsl//tsl/profiler/lib:traceme",
     ],
 )
 
@@ -1212,18 +1212,18 @@ cc_library(
         "@com_google_absl//absl/types:span",
         "@federated_language//federated_language/proto:computation_cc_proto",
         "@org_tensorflow//tensorflow/compiler/jit:xla_cpu_jit",  # buildcleaner: keep # Linking in this dependency ensures that XLA can compile its code for the CPU host.
-        "@org_tensorflow//tensorflow/compiler/xla:literal",
-        "@org_tensorflow//tensorflow/compiler/xla:shape_util",
-        "@org_tensorflow//tensorflow/compiler/xla:xla_data_proto_cc",
-        "@org_tensorflow//tensorflow/compiler/xla:xla_proto_cc",
-        "@org_tensorflow//tensorflow/compiler/xla/client",
-        "@org_tensorflow//tensorflow/compiler/xla/client:client_library",
-        "@org_tensorflow//tensorflow/compiler/xla/client:xla_computation",
-        "@org_tensorflow//tensorflow/compiler/xla/service",
-        "@org_tensorflow//tensorflow/compiler/xla/service:hlo_proto_cc",
-        "@org_tensorflow//tensorflow/compiler/xla/stream_executor:multi_platform_manager",
-        "@org_tensorflow//tensorflow/compiler/xla/stream_executor:platform",
-        "@org_tensorflow//tensorflow/compiler/xla/stream_executor/host:host_platform",  # buildcleaner: keep # Linking in the host platform here ensures that the stream executor can execute on CPU.
+        "@xla//xla:literal",
+        "@xla//xla:shape_util",
+        "@xla//xla:xla_data_proto_cc",
+        "@xla//xla:xla_proto_cc",
+        "@xla//xla/client",
+        "@xla//xla/client:client_library",
+        "@xla//xla/client:xla_computation",
+        "@xla//xla/service",
+        "@xla//xla/service:hlo_proto_cc",
+        "@xla//xla/stream_executor:multi_platform_manager",
+        "@xla//xla/stream_executor:platform",
+        "@xla//xla/stream_executor/host:host_platform",  # buildcleaner: keep # Linking in the host platform here ensures that the stream executor can execute on CPU.
     ],
 )
 
@@ -1251,12 +1251,12 @@ tff_cc_cpu_gpu_test(
         "@federated_language//federated_language/proto:array_cc_proto",
         "@federated_language//federated_language/proto:computation_cc_proto",
         "@federated_language//federated_language/proto:data_type_cc_proto",
-        "@org_tensorflow//tensorflow/compiler/xla:shape_util",
-        "@org_tensorflow//tensorflow/compiler/xla:xla_data_proto_cc",
-        "@org_tensorflow//tensorflow/compiler/xla/client:xla_builder",
-        "@org_tensorflow//tensorflow/compiler/xla/client:xla_computation",
-        "@org_tensorflow//tensorflow/compiler/xla/service:platform_util",
-        "@org_tensorflow//tensorflow/compiler/xla/stream_executor:platform",
+        "@xla//xla:shape_util",
+        "@xla//xla:xla_data_proto_cc",
+        "@xla//xla/client:xla_builder",
+        "@xla//xla/client:xla_computation",
+        "@xla//xla/service:platform_util",
+        "@xla//xla/stream_executor:platform",
     ],
 )
 
@@ -1289,13 +1289,13 @@ cc_test(
         "@federated_language//federated_language/proto:computation_cc_proto",
         "@federated_language//federated_language/proto:data_type_cc_proto",
         "@org_tensorflow//tensorflow/compiler/jit:xla_gpu_jit",  # buildcleaner: keep # Linking in this dependency ensures that XLA can compile its code for the GPU.
-        "@org_tensorflow//tensorflow/compiler/xla:shape_util",
-        "@org_tensorflow//tensorflow/compiler/xla:xla_data_proto_cc",
-        "@org_tensorflow//tensorflow/compiler/xla/client:xla_builder",
-        "@org_tensorflow//tensorflow/compiler/xla/client:xla_computation",
-        "@org_tensorflow//tensorflow/compiler/xla/service:platform_util",
-        "@org_tensorflow//tensorflow/compiler/xla/stream_executor:platform",
-        "@org_tensorflow//tensorflow/compiler/xla/stream_executor/cuda:cuda_platform",  # buildcleaner: keep # Linking in the host platform here ensures that the stream executor can execute on GPU.
+        "@xla//xla:shape_util",
+        "@xla//xla:xla_data_proto_cc",
+        "@xla//xla/client:xla_builder",
+        "@xla//xla/client:xla_computation",
+        "@xla//xla/service:platform_util",
+        "@xla//xla/stream_executor:platform",
+        "@xla//xla/stream_executor/cuda:cuda_platform",  # buildcleaner: keep # Linking in the host platform here ensures that the stream executor can execute on GPU.
     ],
 )
 
@@ -1313,10 +1313,10 @@ cc_library(
         "@federated_language//federated_language/proto:array_cc_proto",
         "@federated_language//federated_language/proto:computation_cc_proto",
         "@federated_language//federated_language/proto:data_type_cc_proto",
-        "@org_tensorflow//tensorflow/compiler/xla:literal",
-        "@org_tensorflow//tensorflow/compiler/xla:shape_util",
-        "@org_tensorflow//tensorflow/compiler/xla:types",
-        "@org_tensorflow//tensorflow/compiler/xla:xla_data_proto_cc",
+        "@xla//xla:literal",
+        "@xla//xla:shape_util",
+        "@xla//xla:types",
+        "@xla//xla:xla_data_proto_cc",
     ],
 )
 
@@ -1336,10 +1336,10 @@ cc_test(
         "@federated_language//federated_language/proto:array_cc_proto",
         "@federated_language//federated_language/proto:computation_cc_proto",
         "@federated_language//federated_language/proto:data_type_cc_proto",
-        "@org_tensorflow//tensorflow/compiler/xla:literal",
-        "@org_tensorflow//tensorflow/compiler/xla:literal_util",
-        "@org_tensorflow//tensorflow/compiler/xla:shape_util",
-        "@org_tensorflow//tensorflow/compiler/xla:types",
-        "@org_tensorflow//tensorflow/compiler/xla:xla_data_proto_cc",
+        "@xla//xla:literal",
+        "@xla//xla:literal_util",
+        "@xla//xla:shape_util",
+        "@xla//xla:types",
+        "@xla//xla:xla_data_proto_cc",
     ],
 )
diff --git tensorflow_federated/cc/core/impl/executors/array_test_utils.h tensorflow_federated/cc/core/impl/executors/array_test_utils.h
index 051b79f20..0a38cbf64 100644
--- tensorflow_federated/cc/core/impl/executors/array_test_utils.h
+++ tensorflow_federated/cc/core/impl/executors/array_test_utils.h
@@ -25,7 +25,7 @@ limitations under the License
 #include "absl/status/statusor.h"
 #include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
-#include "third_party/eigen3/Eigen/Core"
+#include "Eigen/Core"
 #include "federated_language/proto/array.pb.h"
 #include "federated_language/proto/data_type.pb.h"
 
diff --git tensorflow_federated/cc/core/impl/executors/executor.h tensorflow_federated/cc/core/impl/executors/executor.h
index 7e875dc74..88738afa0 100644
--- tensorflow_federated/cc/core/impl/executors/executor.h
+++ tensorflow_federated/cc/core/impl/executors/executor.h
@@ -37,7 +37,7 @@ limitations under the License
 #include "absl/types/span.h"
 #include "absl/utility/utility.h"
 #include "federated_language/proto/computation.pb.h"
-#include "tensorflow/tsl/profiler/lib/traceme.h"
+#include "tsl/profiler/lib/traceme.h"
 #include "tensorflow_federated/cc/core/impl/executors/status_macros.h"
 #include "tensorflow_federated/proto/v0/executor.pb.h"
 
diff --git tensorflow_federated/cc/core/impl/executors/executor_bindings.cc tensorflow_federated/cc/core/impl/executors/executor_bindings.cc
index facf67dd0..cef09bd2c 100644
--- tensorflow_federated/cc/core/impl/executors/executor_bindings.cc
+++ tensorflow_federated/cc/core/impl/executors/executor_bindings.cc
@@ -31,9 +31,9 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/str_cat.h"
-#include "include/grpcpp/impl/channel_interface.h"
-#include "include/grpcpp/security/credentials.h"
-#include "include/grpcpp/support/channel_arguments.h"
+#include "grpcpp/impl/codegen/channel_interface.h"
+#include "grpcpp/security/credentials.h"
+#include "grpcpp/support/channel_arguments.h"
 #include "federated_language/proto/computation.pb.h"
 #include "include/pybind11/cast.h"
 #include "include/pybind11/detail/common.h"
diff --git tensorflow_federated/cc/core/impl/executors/executor_service.cc tensorflow_federated/cc/core/impl/executors/executor_service.cc
index 12fb26078..713bf6dad 100644
--- tensorflow_federated/cc/core/impl/executors/executor_service.cc
+++ tensorflow_federated/cc/core/impl/executors/executor_service.cc
@@ -29,8 +29,8 @@ limitations under the License
 #include "absl/strings/str_join.h"
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/mutex.h"
-#include "include/grpcpp/server_context.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/server_context.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/executor_service.h tensorflow_federated/cc/core/impl/executors/executor_service.h
index 29b35be4a..c0ddb3036 100644
--- tensorflow_federated/cc/core/impl/executors/executor_service.h
+++ tensorflow_federated/cc/core/impl/executors/executor_service.h
@@ -28,8 +28,8 @@ limitations under the License
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/mutex.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/executor_service_test.cc tensorflow_federated/cc/core/impl/executors/executor_service_test.cc
index c881a7fbd..7cc528087 100644
--- tensorflow_federated/cc/core/impl/executors/executor_service_test.cc
+++ tensorflow_federated/cc/core/impl/executors/executor_service_test.cc
@@ -28,8 +28,8 @@ limitations under the License
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "absl/types/span.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/mock_grpc.h tensorflow_federated/cc/core/impl/executors/mock_grpc.h
index c59d9b012..4d68da34b 100644
--- tensorflow_federated/cc/core/impl/executors/mock_grpc.h
+++ tensorflow_federated/cc/core/impl/executors/mock_grpc.h
@@ -20,10 +20,10 @@ limitations under the License
 
 #include "googlemock/include/gmock/gmock.h"
 #include "absl/strings/str_cat.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/security/credentials.h"
-#include "include/grpcpp/security/server_credentials.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/security/credentials.h"
+#include "grpcpp/security/server_credentials.h"
+#include "grpcpp/support/status.h"
 #include "tensorflow_federated/proto/v0/executor.grpc.pb.h"
 #include "tensorflow_federated/proto/v0/executor.pb.h"
 
diff --git tensorflow_federated/cc/core/impl/executors/remote_executor.cc tensorflow_federated/cc/core/impl/executors/remote_executor.cc
index 3ebd1ac67..0f087bc7f 100644
--- tensorflow_federated/cc/core/impl/executors/remote_executor.cc
+++ tensorflow_federated/cc/core/impl/executors/remote_executor.cc
@@ -28,8 +28,8 @@ limitations under the License
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/mutex.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/remote_executor.h tensorflow_federated/cc/core/impl/executors/remote_executor.h
index 619deb5e5..254a530a7 100644
--- tensorflow_federated/cc/core/impl/executors/remote_executor.h
+++ tensorflow_federated/cc/core/impl/executors/remote_executor.h
@@ -18,7 +18,7 @@ limitations under the License
 
 #include <memory>
 
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 #include "tensorflow_federated/proto/v0/executor.grpc.pb.h"
diff --git tensorflow_federated/cc/core/impl/executors/remote_executor_test.cc tensorflow_federated/cc/core/impl/executors/remote_executor_test.cc
index 0f632bf37..e33a61a94 100644
--- tensorflow_federated/cc/core/impl/executors/remote_executor_test.cc
+++ tensorflow_federated/cc/core/impl/executors/remote_executor_test.cc
@@ -31,10 +31,10 @@ limitations under the License
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/notification.h"
 #include "absl/time/time.h"
-#include "include/grpcpp/channel.h"
-#include "include/grpcpp/create_channel.h"
-#include "include/grpcpp/security/credentials.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/channel.h"
+#include "grpcpp/create_channel.h"
+#include "grpcpp/security/credentials.h"
+#include "grpcpp/support/status.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/mock_grpc.h"
diff --git tensorflow_federated/cc/core/impl/executors/status_conversion.cc tensorflow_federated/cc/core/impl/executors/status_conversion.cc
index 43f1b4001..554191229 100644
--- tensorflow_federated/cc/core/impl/executors/status_conversion.cc
+++ tensorflow_federated/cc/core/impl/executors/status_conversion.cc
@@ -18,7 +18,7 @@ limitations under the License
 #include <string>
 
 #include "absl/status/status.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/support/status.h"
 
 namespace tensorflow_federated {
 
diff --git tensorflow_federated/cc/core/impl/executors/status_conversion.h tensorflow_federated/cc/core/impl/executors/status_conversion.h
index 1ec230885..1fbf6a87f 100644
--- tensorflow_federated/cc/core/impl/executors/status_conversion.h
+++ tensorflow_federated/cc/core/impl/executors/status_conversion.h
@@ -17,7 +17,7 @@ limitations under the License
 #define THIRD_PARTY_TENSORFLOW_FEDERATED_CC_CORE_IMPL_EXECUTORS_STATUS_CONVERSION_H_
 
 #include "absl/status/status.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/support/status.h"
 
 namespace tensorflow_federated {
 
diff --git tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.cc tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.cc
index 29627ee07..30824671a 100644
--- tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.cc
+++ tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.cc
@@ -35,8 +35,8 @@ limitations under the License
 #include "absl/strings/str_join.h"
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/mutex.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.h tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.h
index 5a705c7c7..d451bd2ad 100644
--- tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.h
+++ tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.h
@@ -18,7 +18,7 @@ limitations under the License
 
 #include <memory>
 
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 #include "tensorflow_federated/proto/v0/executor.grpc.pb.h"
diff --git tensorflow_federated/cc/core/impl/executors/streaming_remote_executor_test.cc tensorflow_federated/cc/core/impl/executors/streaming_remote_executor_test.cc
index dd2e39828..e9282bcc4 100644
--- tensorflow_federated/cc/core/impl/executors/streaming_remote_executor_test.cc
+++ tensorflow_federated/cc/core/impl/executors/streaming_remote_executor_test.cc
@@ -38,7 +38,7 @@ limitations under the License
 #include "absl/synchronization/notification.h"
 #include "absl/time/time.h"
 #include "absl/types/span.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "federated_language/proto/data_type.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
diff --git tensorflow_federated/cc/core/impl/executors/tensorflow_utils.cc tensorflow_federated/cc/core/impl/executors/tensorflow_utils.cc
index d9c73e603..ecf2d48b0 100644
--- tensorflow_federated/cc/core/impl/executors/tensorflow_utils.cc
+++ tensorflow_federated/cc/core/impl/executors/tensorflow_utils.cc
@@ -27,7 +27,7 @@ limitations under the License
 #include "absl/strings/str_cat.h"
 #include "absl/strings/string_view.h"
 #include "absl/types/span.h"
-#include "third_party/eigen3/Eigen/Core"
+#include "Eigen/Core"
 #include "federated_language/proto/array.pb.h"
 #include "federated_language/proto/data_type.pb.h"
 #include "tensorflow/core/framework/attr_value.pb.h"
diff --git tensorflow_federated/cc/core/impl/executors/tensorflow_utils_test.cc tensorflow_federated/cc/core/impl/executors/tensorflow_utils_test.cc
index 833b80818..c5e537c6a 100644
--- tensorflow_federated/cc/core/impl/executors/tensorflow_utils_test.cc
+++ tensorflow_federated/cc/core/impl/executors/tensorflow_utils_test.cc
@@ -25,7 +25,7 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
-#include "third_party/eigen3/Eigen/Core"
+#include "Eigen/Core"
 #include "federated_language/proto/array.pb.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow/core/framework/numeric_types.h"
diff --git tensorflow_federated/cc/core/impl/executors/xla_executor.cc tensorflow_federated/cc/core/impl/executors/xla_executor.cc
index 0c0f3afc4..975c5eac4 100644
--- tensorflow_federated/cc/core/impl/executors/xla_executor.cc
+++ tensorflow_federated/cc/core/impl/executors/xla_executor.cc
@@ -31,16 +31,16 @@ limitations under the License
 #include "absl/strings/string_view.h"
 #include "absl/types/span.h"
 #include "federated_language/proto/computation.pb.h"
-#include "tensorflow/compiler/xla/client/client.h"
-#include "tensorflow/compiler/xla/client/client_library.h"
-#include "tensorflow/compiler/xla/client/xla_computation.h"
-#include "tensorflow/compiler/xla/literal.h"
-#include "tensorflow/compiler/xla/service/hlo.pb.h"
-#include "tensorflow/compiler/xla/service/service.h"
-#include "tensorflow/compiler/xla/shape.h"
-#include "tensorflow/compiler/xla/stream_executor/platform.h"
-#include "tensorflow/compiler/xla/xla.pb.h"
-#include "tensorflow/compiler/xla/xla_data.pb.h"
+#include "xla/client/client.h"
+#include "xla/client/client_library.h"
+#include "xla/client/xla_computation.h"
+#include "xla/literal.h"
+#include "xla/service/hlo.pb.h"
+#include "xla/service/service.h"
+#include "xla/shape.h"
+#include "xla/stream_executor/platform.h"
+#include "xla/xla.pb.h"
+#include "xla/xla_data.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/status_macros.h"
 #include "tensorflow_federated/cc/core/impl/executors/threading.h"
@@ -49,7 +49,7 @@ limitations under the License
 // clang-format off
 // In TF 2.17 MultiPlatformManager was renamed to PlatformManager. Remove
 // this code when the OSS build gets updated to TF 2.17+.
-#include "tensorflow/compiler/xla/stream_executor/multi_platform_manager.h"
+#include "xla/stream_executor/multi_platform_manager.h"
 namespace stream_executor {
 using PlatformManager = MultiPlatformManager;
 } // namespace stream_executor
diff --git tensorflow_federated/cc/core/impl/executors/xla_executor_test.cc tensorflow_federated/cc/core/impl/executors/xla_executor_test.cc
index 511165a48..c702d4c00 100644
--- tensorflow_federated/cc/core/impl/executors/xla_executor_test.cc
+++ tensorflow_federated/cc/core/impl/executors/xla_executor_test.cc
@@ -34,13 +34,13 @@ limitations under the License
 #include "federated_language/proto/array.pb.h"
 #include "federated_language/proto/computation.pb.h"
 #include "federated_language/proto/data_type.pb.h"
-#include "tensorflow/compiler/xla/client/xla_builder.h"
-#include "tensorflow/compiler/xla/client/xla_computation.h"
-#include "tensorflow/compiler/xla/service/platform_util.h"
-#include "tensorflow/compiler/xla/shape.h"
-#include "tensorflow/compiler/xla/shape_util.h"
-#include "tensorflow/compiler/xla/stream_executor/platform.h"
-#include "tensorflow/compiler/xla/xla_data.pb.h"
+#include "xla/client/xla_builder.h"
+#include "xla/client/xla_computation.h"
+#include "xla/service/platform_util.h"
+#include "xla/shape.h"
+#include "xla/shape_util.h"
+#include "xla/stream_executor/platform.h"
+#include "xla/xla_data.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/array_shape_test_utils.h"
 #include "tensorflow_federated/cc/core/impl/executors/array_test_utils.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/xla_utils.cc tensorflow_federated/cc/core/impl/executors/xla_utils.cc
index 6344f02fa..e0ed0ca53 100644
--- tensorflow_federated/cc/core/impl/executors/xla_utils.cc
+++ tensorflow_federated/cc/core/impl/executors/xla_utils.cc
@@ -27,11 +27,11 @@ limitations under the License
 #include "federated_language/proto/array.pb.h"
 #include "federated_language/proto/computation.pb.h"
 #include "federated_language/proto/data_type.pb.h"
-#include "tensorflow/compiler/xla/literal.h"
-#include "tensorflow/compiler/xla/shape.h"
-#include "tensorflow/compiler/xla/shape_util.h"
-#include "tensorflow/compiler/xla/types.h"
-#include "tensorflow/compiler/xla/xla_data.pb.h"
+#include "xla/literal.h"
+#include "xla/shape.h"
+#include "xla/shape_util.h"
+#include "xla/types.h"
+#include "xla/xla_data.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/status_macros.h"
 
 namespace tensorflow_federated {
diff --git tensorflow_federated/cc/core/impl/executors/xla_utils.h tensorflow_federated/cc/core/impl/executors/xla_utils.h
index af2ebcc51..e7bfcfb76 100644
--- tensorflow_federated/cc/core/impl/executors/xla_utils.h
+++ tensorflow_federated/cc/core/impl/executors/xla_utils.h
@@ -20,8 +20,8 @@ limitations under the License
 #include "federated_language/proto/array.pb.h"
 #include "federated_language/proto/computation.pb.h"
 #include "federated_language/proto/data_type.pb.h"
-#include "tensorflow/compiler/xla/literal.h"
-#include "tensorflow/compiler/xla/shape.h"
+#include "xla/literal.h"
+#include "xla/shape.h"
 
 namespace tensorflow_federated {
 
diff --git tensorflow_federated/cc/core/impl/executors/xla_utils_test.cc tensorflow_federated/cc/core/impl/executors/xla_utils_test.cc
index 229822bd3..fc2f5b078 100644
--- tensorflow_federated/cc/core/impl/executors/xla_utils_test.cc
+++ tensorflow_federated/cc/core/impl/executors/xla_utils_test.cc
@@ -23,17 +23,17 @@ limitations under the License
 #include "googletest/include/gtest/gtest.h"
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
-#include "third_party/eigen3/Eigen/Core"
+#include "Eigen/Core"
 #include "federated_language/proto/array.pb.h"
 #include "federated_language/proto/computation.pb.h"
 #include "federated_language/proto/data_type.pb.h"
-#include "tensorflow/compiler/xla/literal.h"
-#include "tensorflow/compiler/xla/literal_util.h"
-#include "tensorflow/compiler/xla/primitive_util.h"
-#include "tensorflow/compiler/xla/shape.h"
-#include "tensorflow/compiler/xla/shape_util.h"
-#include "tensorflow/compiler/xla/types.h"
-#include "tensorflow/compiler/xla/xla_data.pb.h"
+#include "xla/literal.h"
+#include "xla/literal_util.h"
+#include "xla/primitive_util.h"
+#include "xla/shape.h"
+#include "xla/shape_util.h"
+#include "xla/types.h"
+#include "xla/xla_data.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/array_shape_test_utils.h"
 #include "tensorflow_federated/cc/core/impl/executors/array_test_utils.h"
 #include "tensorflow_federated/cc/testing/protobuf_matchers.h"
diff --git tensorflow_federated/cc/simulation/servers.cc tensorflow_federated/cc/simulation/servers.cc
index c0e45fbbb..d98cf6935 100644
--- tensorflow_federated/cc/simulation/servers.cc
+++ tensorflow_federated/cc/simulation/servers.cc
@@ -23,9 +23,9 @@ limitations under the License
 #include "absl/log/log.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/str_cat.h"
-#include "include/grpcpp/security/server_credentials.h"
-#include "include/grpcpp/server.h"
-#include "include/grpcpp/server_builder.h"
+#include "grpcpp/security/server_credentials.h"
+#include "grpcpp/server.h"
+#include "grpcpp/server_builder.h"
 #include "tensorflow_federated/cc/core/impl/executor_stacks/local_stacks.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/simulation/servers.h tensorflow_federated/cc/simulation/servers.h
index 26b4821cc..da4a2429d 100644
--- tensorflow_federated/cc/simulation/servers.h
+++ tensorflow_federated/cc/simulation/servers.h
@@ -21,8 +21,8 @@ limitations under the License
 #include <memory>
 
 #include "absl/status/statusor.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/security/server_credentials.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/security/server_credentials.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 
diff --git tensorflow_federated/cc/simulation/worker_main.cc tensorflow_federated/cc/simulation/worker_main.cc
index 3bf464d98..9d2bfbe3c 100644
--- tensorflow_federated/cc/simulation/worker_main.cc
+++ tensorflow_federated/cc/simulation/worker_main.cc
@@ -18,7 +18,7 @@ limitations under the License
 
 #include "absl/flags/flag.h"
 #include "absl/flags/parse.h"
-#include "include/grpcpp/security/server_credentials.h"
+#include "grpcpp/security/server_credentials.h"
 #include "tensorflow_federated/cc/simulation/servers.h"
 
 ABSL_FLAG(int32_t, port, 10000, "Port to run the executor service on");
diff --git tensorflow_federated/cc/testing/status_matchers.h tensorflow_federated/cc/testing/status_matchers.h
index aae506006..956dcff27 100644
--- tensorflow_federated/cc/testing/status_matchers.h
+++ tensorflow_federated/cc/testing/status_matchers.h
@@ -30,7 +30,7 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/support/status.h"
 
 namespace tensorflow_federated {
 namespace internal_status {
diff --git tensorflow_federated/proto/v0/BUILD tensorflow_federated/proto/v0/BUILD
index 18c87d39c..c0b88188d 100644
--- tensorflow_federated/proto/v0/BUILD
+++ tensorflow_federated/proto/v0/BUILD
@@ -1,7 +1,8 @@
 load("@com_github_grpc_grpc//bazel:cc_grpc_library.bzl", "cc_grpc_library")
-load("@com_github_grpc_grpc//bazel:python_rules.bzl", "py_grpc_library", "py_proto_library")
+load("@com_github_grpc_grpc//bazel:python_rules.bzl", "py_grpc_library")
 load("@rules_cc//cc:defs.bzl", "cc_proto_library")
 load("@rules_python//python:defs.bzl", "py_library")
+load("@rules_python//python:proto.bzl", "py_proto_library")
 
 package(
     default_applicable_licenses = ["//:package_license"],
