# Fix Bazel dependencies.
#
diff --git tensorflow_federated/cc/core/impl/executor_stacks/executor_stack_bindings.cc tensorflow_federated/cc/core/impl/executor_stacks/executor_stack_bindings.cc
index 6f6253b..2f97a22 100644
--- tensorflow_federated/cc/core/impl/executor_stacks/executor_stack_bindings.cc
+++ tensorflow_federated/cc/core/impl/executor_stacks/executor_stack_bindings.cc
@@ -17,7 +17,7 @@ limitations under the License
 #include <memory>
 #include <vector>
 
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "include/pybind11/detail/common.h"
 #include "include/pybind11/pybind11.h"
 #include "include/pybind11/stl.h"
diff --git tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.cc tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.cc
index 430f168..229dc53 100644
--- tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.cc
+++ tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.cc
@@ -27,7 +27,7 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/str_cat.h"
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/composing_executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.h tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.h
index 6b5e033..eca439e 100644
--- tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.h
+++ tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks.h
@@ -22,7 +22,7 @@ limitations under the License
 #include <vector>
 
 #include "absl/status/statusor.h"
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/composing_executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks_test.cc tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks_test.cc
index 6ec2e83..3b188a3 100644
--- tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks_test.cc
+++ tensorflow_federated/cc/core/impl/executor_stacks/remote_stacks_test.cc
@@ -24,9 +24,9 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "include/grpc/support/time.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/impl/call.h"
-#include "include/grpcpp/security/credentials.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/impl/call.h"
+#include "grpcpp/security/credentials.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/composing_executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/executor_bindings.cc tensorflow_federated/cc/core/impl/executors/executor_bindings.cc
index facf67d..cef09bd 100644
--- tensorflow_federated/cc/core/impl/executors/executor_bindings.cc
+++ tensorflow_federated/cc/core/impl/executors/executor_bindings.cc
@@ -31,9 +31,9 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/str_cat.h"
-#include "include/grpcpp/impl/channel_interface.h"
-#include "include/grpcpp/security/credentials.h"
-#include "include/grpcpp/support/channel_arguments.h"
+#include "grpcpp/impl/codegen/channel_interface.h"
+#include "grpcpp/security/credentials.h"
+#include "grpcpp/support/channel_arguments.h"
 #include "federated_language/proto/computation.pb.h"
 #include "include/pybind11/cast.h"
 #include "include/pybind11/detail/common.h"
diff --git tensorflow_federated/cc/core/impl/executors/executor_service.cc tensorflow_federated/cc/core/impl/executors/executor_service.cc
index 354f0cf..b236632 100644
--- tensorflow_federated/cc/core/impl/executors/executor_service.cc
+++ tensorflow_federated/cc/core/impl/executors/executor_service.cc
@@ -29,8 +29,8 @@ limitations under the License
 #include "absl/strings/str_join.h"
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/mutex.h"
-#include "include/grpcpp/server_context.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/server_context.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/executor_service.h tensorflow_federated/cc/core/impl/executors/executor_service.h
index 29b35be..c0ddb30 100644
--- tensorflow_federated/cc/core/impl/executors/executor_service.h
+++ tensorflow_federated/cc/core/impl/executors/executor_service.h
@@ -28,8 +28,8 @@ limitations under the License
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/mutex.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/executor_service_test.cc tensorflow_federated/cc/core/impl/executors/executor_service_test.cc
index c881a7f..7cc5280 100644
--- tensorflow_federated/cc/core/impl/executors/executor_service_test.cc
+++ tensorflow_federated/cc/core/impl/executors/executor_service_test.cc
@@ -28,8 +28,8 @@ limitations under the License
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "absl/types/span.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/mock_grpc.h tensorflow_federated/cc/core/impl/executors/mock_grpc.h
index c59d9b0..4d68da3 100644
--- tensorflow_federated/cc/core/impl/executors/mock_grpc.h
+++ tensorflow_federated/cc/core/impl/executors/mock_grpc.h
@@ -20,10 +20,10 @@ limitations under the License
 
 #include "googlemock/include/gmock/gmock.h"
 #include "absl/strings/str_cat.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/security/credentials.h"
-#include "include/grpcpp/security/server_credentials.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/security/credentials.h"
+#include "grpcpp/security/server_credentials.h"
+#include "grpcpp/support/status.h"
 #include "tensorflow_federated/proto/v0/executor.grpc.pb.h"
 #include "tensorflow_federated/proto/v0/executor.pb.h"
 
diff --git tensorflow_federated/cc/core/impl/executors/remote_executor.cc tensorflow_federated/cc/core/impl/executors/remote_executor.cc
index ecfe852..b5e8077 100644
--- tensorflow_federated/cc/core/impl/executors/remote_executor.cc
+++ tensorflow_federated/cc/core/impl/executors/remote_executor.cc
@@ -28,8 +28,8 @@ limitations under the License
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/mutex.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/remote_executor.h tensorflow_federated/cc/core/impl/executors/remote_executor.h
index 619deb5..254a530 100644
--- tensorflow_federated/cc/core/impl/executors/remote_executor.h
+++ tensorflow_federated/cc/core/impl/executors/remote_executor.h
@@ -18,7 +18,7 @@ limitations under the License
 
 #include <memory>
 
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 #include "tensorflow_federated/proto/v0/executor.grpc.pb.h"
diff --git tensorflow_federated/cc/core/impl/executors/remote_executor_test.cc tensorflow_federated/cc/core/impl/executors/remote_executor_test.cc
index 0f632bf..e33a61a 100644
--- tensorflow_federated/cc/core/impl/executors/remote_executor_test.cc
+++ tensorflow_federated/cc/core/impl/executors/remote_executor_test.cc
@@ -31,10 +31,10 @@ limitations under the License
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/notification.h"
 #include "absl/time/time.h"
-#include "include/grpcpp/channel.h"
-#include "include/grpcpp/create_channel.h"
-#include "include/grpcpp/security/credentials.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/channel.h"
+#include "grpcpp/create_channel.h"
+#include "grpcpp/security/credentials.h"
+#include "grpcpp/support/status.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 #include "tensorflow_federated/cc/core/impl/executors/mock_grpc.h"
diff --git tensorflow_federated/cc/core/impl/executors/status_conversion.cc tensorflow_federated/cc/core/impl/executors/status_conversion.cc
index 43f1b40..5541912 100644
--- tensorflow_federated/cc/core/impl/executors/status_conversion.cc
+++ tensorflow_federated/cc/core/impl/executors/status_conversion.cc
@@ -18,7 +18,7 @@ limitations under the License
 #include <string>
 
 #include "absl/status/status.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/support/status.h"
 
 namespace tensorflow_federated {
 
diff --git tensorflow_federated/cc/core/impl/executors/status_conversion.h tensorflow_federated/cc/core/impl/executors/status_conversion.h
index 1ec2308..1fbf6a8 100644
--- tensorflow_federated/cc/core/impl/executors/status_conversion.h
+++ tensorflow_federated/cc/core/impl/executors/status_conversion.h
@@ -17,7 +17,7 @@ limitations under the License
 #define THIRD_PARTY_TENSORFLOW_FEDERATED_CC_CORE_IMPL_EXECUTORS_STATUS_CONVERSION_H_
 
 #include "absl/status/status.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/support/status.h"
 
 namespace tensorflow_federated {
 
diff --git tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.cc tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.cc
index 4701c29..2b7c1aa 100644
--- tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.cc
+++ tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.cc
@@ -35,8 +35,8 @@ limitations under the License
 #include "absl/strings/str_join.h"
 #include "absl/strings/string_view.h"
 #include "absl/synchronization/mutex.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/support/status.h"
 #include "federated_language/proto/computation.pb.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.h tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.h
index 5a705c7..d451bd2 100644
--- tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.h
+++ tensorflow_federated/cc/core/impl/executors/streaming_remote_executor.h
@@ -18,7 +18,7 @@ limitations under the License
 
 #include <memory>
 
-#include "include/grpcpp/grpcpp.h"
+#include "grpcpp/grpcpp.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 #include "tensorflow_federated/proto/v0/executor.grpc.pb.h"
diff --git tensorflow_federated/cc/core/impl/executors/streaming_remote_executor_test.cc tensorflow_federated/cc/core/impl/executors/streaming_remote_executor_test.cc
index d1c6a0a..35aafda 100644
--- tensorflow_federated/cc/core/impl/executors/streaming_remote_executor_test.cc
+++ tensorflow_federated/cc/core/impl/executors/streaming_remote_executor_test.cc
@@ -38,7 +38,7 @@ limitations under the License
 #include "absl/synchronization/notification.h"
 #include "absl/time/time.h"
 #include "absl/types/span.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/support/status.h"
 #include "google/protobuf/text_format.h"
 #include "federated_language/proto/computation.pb.h"
 #include "federated_language/proto/data_type.pb.h"
diff --git tensorflow_federated/cc/simulation/servers.cc tensorflow_federated/cc/simulation/servers.cc
index c0e45fb..d98cf69 100644
--- tensorflow_federated/cc/simulation/servers.cc
+++ tensorflow_federated/cc/simulation/servers.cc
@@ -23,9 +23,9 @@ limitations under the License
 #include "absl/log/log.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/str_cat.h"
-#include "include/grpcpp/security/server_credentials.h"
-#include "include/grpcpp/server.h"
-#include "include/grpcpp/server_builder.h"
+#include "grpcpp/security/server_credentials.h"
+#include "grpcpp/server.h"
+#include "grpcpp/server_builder.h"
 #include "tensorflow_federated/cc/core/impl/executor_stacks/local_stacks.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
diff --git tensorflow_federated/cc/simulation/servers.h tensorflow_federated/cc/simulation/servers.h
index 26b4821..da4a242 100644
--- tensorflow_federated/cc/simulation/servers.h
+++ tensorflow_federated/cc/simulation/servers.h
@@ -21,8 +21,8 @@ limitations under the License
 #include <memory>
 
 #include "absl/status/statusor.h"
-#include "include/grpcpp/grpcpp.h"
-#include "include/grpcpp/security/server_credentials.h"
+#include "grpcpp/grpcpp.h"
+#include "grpcpp/security/server_credentials.h"
 #include "tensorflow_federated/cc/core/impl/executors/cardinalities.h"
 #include "tensorflow_federated/cc/core/impl/executors/executor.h"
 
diff --git tensorflow_federated/cc/simulation/worker_main.cc tensorflow_federated/cc/simulation/worker_main.cc
index 3bf464d..9d2bfbe 100644
--- tensorflow_federated/cc/simulation/worker_main.cc
+++ tensorflow_federated/cc/simulation/worker_main.cc
@@ -18,7 +18,7 @@ limitations under the License
 
 #include "absl/flags/flag.h"
 #include "absl/flags/parse.h"
-#include "include/grpcpp/security/server_credentials.h"
+#include "grpcpp/security/server_credentials.h"
 #include "tensorflow_federated/cc/simulation/servers.h"
 
 ABSL_FLAG(int32_t, port, 10000, "Port to run the executor service on");
diff --git tensorflow_federated/cc/testing/status_matchers.h tensorflow_federated/cc/testing/status_matchers.h
index aae5060..956dcff 100644
--- tensorflow_federated/cc/testing/status_matchers.h
+++ tensorflow_federated/cc/testing/status_matchers.h
@@ -30,7 +30,7 @@ limitations under the License
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
-#include "include/grpcpp/support/status.h"
+#include "grpcpp/support/status.h"
 
 namespace tensorflow_federated {
 namespace internal_status {
