diff --git tensorflow_federated/python/core/impl/executors/BUILD tensorflow_federated/python/core/impl/executors/BUILD
index 03c597916..2e5dba13f 100644
--- tensorflow_federated/python/core/impl/executors/BUILD
+++ tensorflow_federated/python/core/impl/executors/BUILD
@@ -29,6 +29,7 @@ py_library(
     tags = [
         "nokokoro",  # b/193543632: C++ execution is not fully supported in OSS.
     ],
+    visibility = ["//visibility:public"],
     deps = [
         ":executor_bindings",
         ":executors_errors",
