# Update visibility for the `cpp_to_python_executor` target.
#
diff --git tensorflow_federated/python/core/impl/executors/BUILD tensorflow_federated/python/core/impl/executors/BUILD
index 693ea3a..85f4468 100644
--- tensorflow_federated/python/core/impl/executors/BUILD
+++ tensorflow_federated/python/core/impl/executors/BUILD
@@ -49,6 +49,7 @@ py_test(
 
 py_library(
     name = "cpp_to_python_executor",
+    visibility = ["//visibility:public"],
     srcs = ["cpp_to_python_executor.py"],
     deps = [
         ":executor_bindings",
