diff --git tensorflow_federated/python/common_libs/BUILD tensorflow_federated/python/common_libs/BUILD
index 2a52b4516..4b033aa00 100644
--- tensorflow_federated/python/common_libs/BUILD
+++ tensorflow_federated/python/common_libs/BUILD
@@ -84,6 +84,5 @@ py_library(
     srcs = ["structure.py"],
     deps = [
         ":deprecation",
-        "@federated_language//federated_language/common_libs:structure",
     ],
 )
